src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -98,6 +100,28 @@ protected function redirectToRoute($route, array $parameters = array(), $status
     }
 
     /**
+     * Returns a JsonResponse that uses the serializer component if enabled, or json_encode.
+     *
+     * @param mixed $data    The response data
+     * @param int   $status  The status code to use for the Response
+     * @param array $headers Array of extra headers to add
+     * @param array $context Context to pass to serializer when using serializer component
+     *
+     * @return JsonResponse
+     */
+    protected function json($data, $status = 200, $headers = array(), $context = array())
+    {
+        if ($this->container->has('serializer')) {
```

Comment:
I'm not a fan of having 2 different behaviors depending of the configuration. What do you think about splitting this method in 2 separates methods and throw an error in the method using the serializer if it is not available?